include/string_utils.h

```diff
@@ -363,7 +363,7 @@ std::optional<float> parse_percentage_with_percent_sign(const std::string_view s
 std::optional<float> parse_percentage_with_optional_percent_sign(const std::string_view s);
 
 template <typename... Args>
-std::string format_string(const std::string& format, const Args&... args) noexcept
+std::string format_str(const std::string& format, const Args&... args) noexcept
```

Comment:
Could you explain what is the rationale behind this change?

Reply:
I put a short note in the commit message:

```log
C++20 provides std::format_string when <format> is included, triggering a compiler error on MSClang.
```

So the <format> header dumps [`format_string`](https://en.cppreference.com/w/cpp/utility/format/basic_format_string) into the global namespace, and MSClang cannot compile it. We shouldn't be having the same name as built-ins.